fearless/Modules/Staking/Analytics/AnalyticsValidators/AnalyticsValidatorsInteractor.swift

```diff
@@ -0,0 +1,217 @@
+    weak var presenter: AnalyticsValidatorsInteractorOutputProtocol!
+    let selectedAddress: AccountAddress
+    let substrateProviderFactory: SubstrateDataProviderFactoryProtocol
+    let singleValueProviderFactory: SingleValueProviderFactoryProtocol
+    let identityOperationFactory: IdentityOperationFactoryProtocol
+    let operationManager: OperationManagerProtocol
+    let engine: JSONRPCEngine
+    let runtimeService: RuntimeCodingServiceProtocol
+    let storageRequestFactory: StorageRequestFactoryProtocol
+    let chain: Chain
+
+    private var stashItemProvider: StreamableProvider<StashItem>?
+    private var nominationProvider: AnyDataProvider<DecodedNomination>?
+    private var identitiesByAddress: [AccountAddress: AccountIdentity]?
+
+    init(
+        selectedAddress: AccountAddress,
+        substrateProviderFactory: SubstrateDataProviderFactoryProtocol,
+        singleValueProviderFactory: SingleValueProviderFactoryProtocol,
+        identityOperationFactory: IdentityOperationFactoryProtocol,
+        operationManager: OperationManagerProtocol,
+        engine: JSONRPCEngine,
+        runtimeService: RuntimeCodingServiceProtocol,
+        storageRequestFactory: StorageRequestFactoryProtocol,
+        chain: Chain
+    ) {
+        self.selectedAddress = selectedAddress
+        self.substrateProviderFactory = substrateProviderFactory
+        self.singleValueProviderFactory = singleValueProviderFactory
+        self.identityOperationFactory = identityOperationFactory
+        self.operationManager = operationManager
+        self.engine = engine
+        self.runtimeService = runtimeService
+        self.storageRequestFactory = storageRequestFactory
+        self.chain = chain
+    }
+
+    private func fetchValidatorIdentity(accountIds: [AccountId]) {
+        let operation = identityOperationFactory.createIdentityWrapper(
+            for: { accountIds },
+            engine: engine,
+            runtimeService: runtimeService,
+            chain: chain
+        )
+        operation.targetOperation.completionBlock = { [weak self] in
+            DispatchQueue.main.async {
+                do {
+                    let identitiesByAddress = try operation.targetOperation.extractNoCancellableResultData()
+                    self?.presenter.didReceive(identitiesByAddressResult: .success(identitiesByAddress))
+                } catch {
+                    self?.presenter.didReceive(identitiesByAddressResult: .failure(error))
+                }
+            }
+        }
+        operationManager.enqueue(operations: operation.allOperations, in: .transient)
+    }
+
+    private func createChainHistoryRangeOperationWrapper(
+        codingFactoryOperation: BaseOperation<RuntimeCoderFactoryProtocol>
+    ) -> CompoundOperationWrapper<ChainHistoryRange> {
+        let keyFactory = StorageKeyFactory()
+
+        let currentEraWrapper: CompoundOperationWrapper<[StorageResponse<StringScaleMapper<UInt32>>]> =
```

Comment:
I think we should build history range as `[submittedIn + 1; currentEra]` where submittedIn in which current set of validators was submitted by a nominator. `submittedIn` can be obtained from `Nomination` that is already received via subscription